File: bakkBenchmark.JavaScript/src/DNAGenerator.ts
class DNAGenerator
{
    private static readonly IM = 139968;
    private static readonly IA = 3877;
    private static readonly IC = 29573;
    private static seed = 42;

    public static generate(n: number, length: number): string[]
    {
        var res = new Array<string>();

        for (let i = 0; i < n; i++)
        {
            var arr = new Array<string>(length);

            for (let j = 0; j < length; j++)
            {
                arr.push(DNAGenerator.getRandomCode());
            }

            res[i] = arr.join('');
        }

        return res;
    }

    private static getRandomCode(): string
    {
        var r = DNAGenerator.random();

        if (r < 0.3 * DNAGenerator.IM) {
            return 'A';
        } else if (r < 0.5 * DNAGenerator.IM) {
            return 'C';
        } else if (r < 0.7 * DNAGenerator.IM) {
            return 'G';
        } else {
            return 'T';
        }
    }

    private static random(): number
    {
        DNAGenerator.seed = (DNAGenerator.seed * DNAGenerator.IA + DNAGenerator.IC) % DNAGenerator.IM;
        return DNAGenerator.seed;
    }
}

File: bakkBenchmark.DotNet/DNAGenerator.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace BakkBenchmark.DotNet
{
    public static class DNAGenerator
    {
        private const int IM = 139968;
        private const int IA = 3877;
        private const int IC = 29573;
        private static int seed = 42;

        public static string[] Generate(int n, int length)
        {
            var res = new string[n];

            for (int i = 0; i < n; i++)
            {
                var str = new char[length];

                for (int j = 0; j < length; j++)
                {
                    str[j] = GetRandomCode();
                }

                res[i] = new string(str);
            }

            return res;
        }

        private static char GetRandomCode()
        {
            var r = Random();

            if (r < 0.25 * IM)
                return 'A';
            else if (r < 0.5 * IM)
                return 'C';
            else if (r < 0.75 * IM)
                return 'G';
            else
                return 'T';
        }

        private static int Random()
        {
            seed = (seed * IA + IC) % IM;
            return seed;
        }
    }
}